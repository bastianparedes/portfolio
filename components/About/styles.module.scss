.main {
  display: grid;
  gap: 80px;
  grid-template-columns: 1fr 1fr;
}

.mainLeft {
  &__p {
    text-align: justify;
    line-height: 40px;
    font-size: 1.2rem;
  }
}

.buttonCv {
  display: block;
  font-size: 1.5rem;
  margin: 0 auto;
  padding: 1rem;
  cursor: pointer;
  background-color: rgb(103, 212, 255);
  border: black 1px solid;

  &__a {
    text-decoration: none;
    color: rgb(0, 0, 0);
    cursor: inherit;
  }
}

.mainRight {
  &__h2 {
    text-align: center;
  }
}

.skillsContainer {
  position: relative;
  padding-top: 0.9rem;
  display: grid;
  place-items: center;
  gap: 2.5rem 6rem;
  grid-template-columns: repeat(auto-fit, minmax(4rem, 1fr));
}

@media screen and (max-width: 768px) {
  .main {
    grid-template-columns: 1fr;
  }
  .skillsContainer {
    gap: 2.5rem 2.5rem;
    grid-template-columns: repeat(auto-fit, minmax(5rem, 1fr));
  }
}

@media screen and (max-width: 480px) {
  .skillsContainer {
    gap: 2.5rem 1rem;
    grid-template-columns: repeat(auto-fit, minmax(5rem, 1fr));
  }
}
