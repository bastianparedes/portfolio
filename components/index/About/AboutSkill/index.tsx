import React from 'react';

import Image from 'next/legacy/image';

import styles from './styles.module.scss';

interface SkillProps {
  src: string;
  name: string;
}

const AboutSkill = (props: SkillProps): JSX.Element => {
  return (
    <div className={styles.skillCard}>
      <div className={styles.skillImageContainer}>
        <Image
          alt={props.name}
          layout="fill"
          objectFit="contain"
          placeholder="blur"
          src={'/images/skills/' + props.src}
        />
      </div>
      <p className={styles.skillName}>{props.name}</p>
    </div>
  );
};

export default AboutSkill;
