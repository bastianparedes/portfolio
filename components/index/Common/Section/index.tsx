import React from 'react';
import styles from './styles.module.scss';

interface SectionProps {
  id: string;
  title: string;
  children?: React.ReactNode;
}

const Section = (props: SectionProps): JSX.Element => {
  return (
    <section id={props.id} className={styles.section}>
      <h3 className={styles.title}>{props.title}</h3>
      {props.children}
    </section>
  );
};

export default Section;
