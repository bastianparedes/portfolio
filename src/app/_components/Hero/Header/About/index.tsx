'use client';

import { useState } from 'react';
import Modal from '../../../Modal';
import { useTranslationContext } from '../../../../_contexts/translation';

const Component = () => {
  const [showContent, setShowContent] = useState(false);
  const { translation } = useTranslationContext();
  const openModal = () => setShowContent(true);
  const closeModal = () => setShowContent(false);

  const yearsExperience = new Date().getFullYear() - 2020;

  return (
    <>
      <button onClick={openModal}>{translation.about.button}</button>
      {showContent && (
        <Modal closeModal={closeModal}>
          <div className="flex flex-col gap-5">
            <p>
              {translation.about.paragraph_1.part_1} {yearsExperience}{' '}
              {translation.about.paragraph_1.part_2}
            </p>
            <p>{translation.about.paragraph_2}</p>
          </div>
        </Modal>
      )}
    </>
  );
};

export default Component;
