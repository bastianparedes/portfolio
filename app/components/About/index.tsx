import React from 'react';

import Image from 'next/image';

import images from './skillImages';
import styles from './styles.module.scss';
import constants from '../../../config/constants';
import { Section } from '../Common';

const About = (): JSX.Element => {
  return (
    <Section
      id={constants.NAV.BUTTONS.ABOUT.ID}
      title={constants.ABOUT.SECTION.TITLE}
    >
      <div className={styles.main}>
        <div className={styles.mainLeft}>
          <h2>{constants.ABOUT.DESCRIPTION_PART_1}</h2>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_2}
          </p>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_3}
          </p>
          <button className={styles.buttonCv}>
            <a
              className={styles.buttonCv__a}
              href={constants.ABOUT.CV.LINK}
              rel="noreferrer noopener"
              target="_blank"
            >
              {constants.ABOUT.CV.MESSAGE}
            </a>
          </button>
        </div>
        <div>
          <h2 className={styles.mainRight__h2}>
            {constants.ABOUT.SKILLS.TITLE}
          </h2>
          <div className={styles.skillsContainer}>
            {Object.keys(images).map((skill) => (
              <div className={styles.skillCard} key={skill}>
                <div className={styles.skillImageContainer}>
                  <Image
                    alt={skill}
                    placeholder="blur"
                    src={images[skill as keyof typeof images]}
                    style={{
                      height: '100%',
                      objectFit: 'contain',
                      width: '100%'
                    }}
                  />
                </div>
                <p className={styles.skillName}>{skill}</p>
              </div>
            ))}
          </div>
        </div>
      </div>
    </Section>
  );
};

export default About;
