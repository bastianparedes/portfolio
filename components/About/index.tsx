import React from 'react'
import styles from './styles.module.css'
import { Section } from '../Common'
import constants from '../../config/constants.json'
import AboutSkill from './AboutSkill'

const About = (): JSX.Element => {
  return (
    <Section
      id={constants.NAV.BUTTONS.ABOUT.ID}
      title={constants.ABOUT.SECTION.TITLE}
    >
      <div className={styles.main}>
        <div className={styles.mainLeft}>
          <h2 className={styles.mainLeft__h2}>
            {constants.ABOUT.DESCRIPTION_PART_1}
          </h2>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_2}
          </p>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_3}
          </p>
          <button className={styles.buttonCv}>
            <a
              className={styles.buttonCv__a}
              href={constants.ABOUT.CV.LINK}
              target="_blank"
              rel="noreferrer noopener"
            >
              {constants.ABOUT.CV.MESSAGE}
            </a>
          </button>
        </div>
        <div>
          <h2 className={styles.mainRight__h2}>
            {constants.ABOUT.SKILLS.TITLE}
          </h2>
          <div className={styles.skillsContainer}>
            {constants.ABOUT.SKILLS.LIST.map((skill) => (
              <AboutSkill key={skill.NAME} src={skill.SRC} name={skill.NAME} />
            ))}
          </div>
        </div>
      </div>
    </Section>
  )
}

export default About
