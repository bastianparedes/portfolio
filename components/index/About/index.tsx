import React from 'react';

import constants from '../../../config/constants';
import { Section } from '../Common';
import AboutSkill from './AboutSkill';
import styles from './styles.module.scss';

const About = (): JSX.Element => {
  return (
    <Section
      id={constants.NAV.BUTTONS.ABOUT.ID}
      title={constants.ABOUT.SECTION.TITLE}
    >
      <div className={styles.main}>
        <div className={styles.mainLeft}>
          <h2>{constants.ABOUT.DESCRIPTION_PART_1}</h2>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_2}
          </p>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_3}
          </p>
          <button className={styles.buttonCv}>
            <a
              className={styles.buttonCv__a}
              href={constants.ABOUT.CV.LINK}
              target="_blank"
              rel="noreferrer noopener"
            >
              {constants.ABOUT.CV.MESSAGE}
            </a>
          </button>
        </div>
        <div>
          <h2 className={styles.mainRight__h2}>
            {constants.ABOUT.SKILLS.TITLE}
          </h2>
          <div className={styles.skillsContainer}>
            {constants.ABOUT.SKILLS.LIST.map((skill) => (
              <AboutSkill key={skill.NAME} src={skill.SRC} name={skill.NAME} />
            ))}
          </div>
        </div>
      </div>
    </Section>
  );
};

export default About;
