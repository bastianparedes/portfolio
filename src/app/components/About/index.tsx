import React from 'react';

import Image from 'next/image';

import skills from './skills';
import styles from './styles.module.scss';
import constants from '../../../../config/constants';
import { Section } from '../Common';

const About = () => {
  return (
    <Section
      id={constants.NAV.BUTTONS.ABOUT.ID}
      title={constants.ABOUT.SECTION.TITLE}
    >
      <div className={styles.main}>
        <div className={styles.mainLeft}>
          <h2>{constants.ABOUT.DESCRIPTION_PART_1}</h2>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_2}
          </p>
          <p className={styles.mainLeft__p}>
            {constants.ABOUT.DESCRIPTION_PART_3}
          </p>
          <button className={styles.buttonCv}>
            <a
              className={styles.buttonCv__a}
              href={constants.ABOUT.CV.LINK}
              rel="noreferrer noopener"
              target="_blank"
            >
              {constants.ABOUT.CV.MESSAGE}
            </a>
          </button>
        </div>
        <div>
          <h2 className={styles.mainRight__h2}>
            {constants.ABOUT.SKILLS.TITLE}
          </h2>
          <div className={styles.skillsContainer}>
            {skills.map((skill) => (
              <div className={styles.skillCard} key={skill.name}>
                <div className={styles.skillImageContainer}>
                  <Image
                    alt={skill.name}
                    placeholder="blur"
                    src={skill.image}
                    style={{
                      height: '100%',
                      objectFit: 'contain',
                      width: '100%'
                    }}
                  />
                </div>
                <p className={styles.skillName}>{skill.name}</p>
              </div>
            ))}
          </div>
        </div>
      </div>
    </Section>
  );
};

export default About;
