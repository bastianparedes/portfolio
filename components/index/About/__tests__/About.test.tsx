import React from 'react';

import { render } from '@testing-library/react';

import About from '..';
import { Section } from '../../Common';

jest.mock('../../Common');
jest.mock('../AboutSkill');

describe('<About />', () => {
  (Section as jest.Mock).mockImplementation(({ children }) => <>{children}</>);
  it('should render', () => {
    const { container } = render(<About />);
    expect(container).toMatchSnapshot();
  });
});
