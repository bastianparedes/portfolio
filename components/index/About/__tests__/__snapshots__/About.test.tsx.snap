// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should render 1`] = `
<div>
  <section
    class="section"
    id="about"
  >
    <h3
      class="title"
    >
      About me
    </h3>
    <div
      class="main"
    >
      <div
        class="mainLeft"
      >
        <h2>
          Soy Bastián Paredes,
        </h2>
        <p
          class="mainLeft__p"
        >
          Un Front End Developer con experiencia construyendo sitios web con responsive design que conecten front-end y back-end. Me especializo en el uso de JavaScript, React.js y Next.js.
        </p>
        <p
          class="mainLeft__p"
        >
          Siempre busco mejorar mis habilidades desarrollando proyectos propios a la par de mi trabajo. Estoy constantemente estudiando acerca de las nuevas tecnologías para actualizar mi repertorio de herramientas para que los códigos de mis creaciones sean fáciles de leer, seguros y eficientes.
        </p>
        <button
          class="buttonCv"
        >
          <a
            class="buttonCv__a"
            href="/api/cv"
            rel="noreferrer noopener"
            target="_blank"
          >
            Ver currículum
          </a>
        </button>
      </div>
      <div>
        <h2
          class="mainRight__h2"
        >
          Skills
        </h2>
        <div
          class="skillsContainer"
        >
          <div
            class="skillCard"
          >
            <div
              class="skillImageContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="HTML 5"
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
                <noscript />
              </span>
            </div>
            <p
              class="skillName"
            >
              HTML 5
            </p>
          </div>
          <div
            class="skillCard"
          >
            <div
              class="skillImageContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="CSS 3"
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
                <noscript />
              </span>
            </div>
            <p
              class="skillName"
            >
              CSS 3
            </p>
          </div>
          <div
            class="skillCard"
          >
            <div
              class="skillImageContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="JavaScript"
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
                <noscript />
              </span>
            </div>
            <p
              class="skillName"
            >
              JavaScript
            </p>
          </div>
          <div
            class="skillCard"
          >
            <div
              class="skillImageContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="TypeScript"
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
                <noscript />
              </span>
            </div>
            <p
              class="skillName"
            >
              TypeScript
            </p>
          </div>
          <div
            class="skillCard"
          >
            <div
              class="skillImageContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="React.js"
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
                <noscript />
              </span>
            </div>
            <p
              class="skillName"
            >
              React.js
            </p>
          </div>
          <div
            class="skillCard"
          >
            <div
              class="skillImageContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="Next.js"
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
                <noscript />
              </span>
            </div>
            <p
              class="skillName"
            >
              Next.js
            </p>
          </div>
          <div
            class="skillCard"
          >
            <div
              class="skillImageContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="Python 3"
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
                <noscript />
              </span>
            </div>
            <p
              class="skillName"
            >
              Python 3
            </p>
          </div>
          <div
            class="skillCard"
          >
            <div
              class="skillImageContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="MySql"
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
                <noscript />
              </span>
            </div>
            <p
              class="skillName"
            >
              MySql
            </p>
          </div>
          <div
            class="skillCard"
          >
            <div
              class="skillImageContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="MongoDB"
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                />
                <noscript />
              </span>
            </div>
            <p
              class="skillName"
            >
              MongoDB
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
